orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForAllInstancesNotUpTaskSpec.groovy

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.servergroup
+
+import com.netflix.spinnaker.orca.pipeline.model.Pipeline
+import com.netflix.spinnaker.orca.pipeline.model.PipelineStage
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class WaitForAllInstancesNotUpTaskSpec extends Specification {
+  @Subject task = new WaitForAllInstancesNotUpTask()
+
+  @Unroll
+  void "should succeed as #hasSucceeded based on instance providers #healthProviderNames for instances #instances"() {
```

Comment:
A copy of what we have for WaitforAllInstancesDownTask with:
- `platform` tests removed.
- success if a `healthProviderName` is specified but doesn't exist on an instance